### PR TITLE
PLT-5436: Render channel links in the channel header

### PR DESCRIPTION
**Pull request by panditsavitags**
#### Summary
Channel Header is now linked to the specific channel if the text following the (~) matches any of the display_names. 

#### Ticket Link
[issue-7257](https://github.com/mattermost/mattermost-server/issues/7257)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [ ] Has server changes (please link)
- [ ] Has redux changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

**Comment by jasonblais** [comment]
@panditsavitags Looks like there's a build failure. Can you help take a look?

**Comment by panditsavitags** [comment]
@jasonblais , Please can you paste the error? I am not able to check the details of the build faliure.

**Comment by hmhealey** [comment]
@panditsavitags It's just a style issue
```
$ eslint --ext ".jsx" --ignore-pattern node_modules --quiet .

/home/jenkins/workspace/mattermost-webapp_PR-387-2HIGC7TBLEDX57RY5YNGAT5NIC75Q437465WSNID364XJDUTNCXQ/mattermost-webapp/components/user_settings/user_settings_display.jsx
  462:13  error  'teammateNameSection' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by panditsavitags** [comment]
@hmhealey , I have fixed the style error, please can you check what went wrong now?

File: tests/components/__snapshots__/user_settings_display.test.jsx.snap [hunk 1]
```diff
@@ -340,6 +340,143 @@ exports[`components/user_settings/UserSettingsDisplay should match snapshot, clo
                   id="user.settings.display.preferTime"
                   values={Object {}}
                 />
+<<<<<<< HEAD
```
Comment (sudheerDev) [inline tests/components/__snapshots__/user_settings_display.test.jsx.snap]: You would have to resolve conflicts here in snapshots as well. Choose head or your commit changes appropriately.

Comment (panditsavitags) [reply tests/components/__snapshots__/user_settings_display.test.jsx.snap]: @sudheerDev , I have raised another PR with commits, Please check https://github.com/mattermost/mattermost-webapp/pull/398#issuecomment-348964883.
This is a closed PR

Comment (sudheerDev) [reply tests/components/__snapshots__/user_settings_display.test.jsx.snap]: @panditsavitags Oops. I just happened to see this commit So, was just pitching in.  I am not a reviewer. Let me see if I can help though :)

Comment (panditsavitags) [reply tests/components/__snapshots__/user_settings_display.test.jsx.snap]: :) thanks

**Pull request closed without merging**